### PR TITLE
[Admin][Core] Fix calculation issues on order show

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | yes
| BC breaks?      | yes
| Related tickets | 
| License         | MIT

This PR some issues on new admin order show page and introduce new logic of shipping promotions:
- use methods introduced in [this PR](https://github.com/Sylius/Sylius/pull/5084) in order show page on *admin* module
- introduced ``order shipping promotion`` adjustment, that replaces ``order promotion`` usage in ``shipping discount`` action
- fix shipping taxes applying (shipping promotions has not been included)

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Not including neutral tax adjustments in the tax total is not entirely correct. Even if the taxes are included in price, we need to display the total tax amount correctly - including that tax.

File: features/order/managing_orders/seeing_order_item_taxes.feature [hunk 1]
```diff
@@ -25,3 +28,12 @@ Feature: Seeing taxes of order items
         Then the order's shipping total should be "€0.00"
         And the order's tax total should be "€32.20"
         And the order's total should be "€252.20"
+
+    @ui
+    Scenario: Seeing included in price taxes of order items are not counted in taxes total
```
Comment (michalmarcinkowski) [inline features/order/managing_orders/seeing_order_item_taxes.feature]: Please extract this to a separate feature file.

1. We should not mix included and excluded tax rates within one channel
2. The background becomes too big

File: src/Sylius/Component/Core/Model/OrderItemUnit.php [hunk 2]
```diff
@@ -146,7 +146,9 @@ public function getTaxTotal()
         $taxTotal = 0;
 
         foreach ($this->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT) as $taxAdjustment) {
-            $taxTotal += $taxAdjustment->getAmount();
+            if (!$taxAdjustment->isNeutral()) {
+                $taxTotal += $taxAdjustment->getAmount();
+            }
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Model/OrderItemUnit.php]: Why changing this?

File: src/Sylius/Component/Core/Taxation/Applicator/OrderShipmentTaxesApplicator.php [hunk 3]
```diff
@@ -75,7 +75,12 @@ public function apply(OrderInterface $order, ZoneInterface $zone)
         }
 
         $lastShippingAdjustment = $shippingAdjustments->last();
-        $taxAmount = $this->calculator->calculate($lastShippingAdjustment->getAmount(), $taxRate);
+        $shippingPrice =
+            $lastShippingAdjustment->getAmount()
+            + $order->getAdjustmentsTotal(AdjustmentInterface::ORDER_SHIPPING_PROMOTION_ADJUSTMENT)
+        ;
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Taxation/Applicator/OrderShipmentTaxesApplicator.php]: How about `$shippingPrice = $order->getShippingTotal()`?

**Comment by michalmarcinkowski** [comment]
We also should have scenarios for aggregating order promotion and tax adjustments.

**Comment by Zales0123** [comment]
@michalmarcinkowski Totally agreed 👍  However, maybe we could introduce them in separate PR, as it's similar but still different problem and we don't want to delay this PR?

**Comment by pjedrzejewski** [comment]
Thanks Mateusz! 👍

**Pull request merged**